plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -260,58 +213,63 @@ func TestWriteWithType(t *testing.T) {
-					IngestionType:   ingestionType,
-					MetricsGrouping: testCase.metricsGrouping,
-					TableName:       tableName,
-					CreateTables:    testCase.createTables,
-					kustoClient:     fakeClient,
-					metricIngestors: map[string]ingest.Ingestor{
-						tableName: mockIngestor,
+					Config: adx_common.Config{
+						Endpoint:        "someendpoint",
+						Database:        "databasename",
+						IngestionType:   ingestionType,
+						MetricsGrouping: testCase.metricsGrouping,
+						TableName:       tableName,
+						CreateTables:    testCase.createTables,
 					},
 					serializer: serializer,
+					Log:        testutil.Logger{},
+					Client:     &adx_common.Client{},
 				}
+				plugin.Client.SetKustoClient(fakeClient)
+				plugin.Client.SetConfig(&plugin.Config)
+				plugin.Client.SetLogger(plugin.Log)
+				plugin.Client.SetMetricsIngestors(map[string]ingest.Ingestor{
+					tableName: mockIngestor,
+				})
+
 				err := plugin.Write(testCase.inputMetric)
 				if testCase.expectedWriteError != "" {
 					require.EqualError(t, err, testCase.expectedWriteError)
 					continue
 				}
 				require.NoError(t, err)
-				createdIngestor := plugin.metricIngestors[tableName]
-				if testCase.metricsGrouping == singleTable {
-					createdIngestor = plugin.metricIngestors[tableName]
-				}
+
 				records := mockIngestor.records[0] // the first element
-				require.NotNil(t, createdIngestor)
 				require.JSONEq(t, jsonValue, records)
 			}
 		})
 	}
 }
 
-func TestInitBlankEndpointData(t *testing.T) {
+func TestInit(t *testing.T) {
 	plugin := AzureDataExplorer{
-		Log:             testutil.Logger{},
-		kustoClient:     kusto.NewMockClient(),
-		metricIngestors: map[string]ingest.Ingestor{},
+		Log:    testutil.Logger{},
+		Client: &adx_common.Client{},
+		Config: adx_common.Config{
+			Endpoint: "someendpoint",
+		},
 	}
 
-	errorInit := plugin.Init()
-	require.Error(t, errorInit)
-	require.Equal(t, "endpoint configuration cannot be empty", errorInit.Error())
+	err := plugin.Init()
+	require.NoError(t, err)
 }
 
-func TestQueryConstruction(t *testing.T) {
-	const tableName = "mytable"
-	const expectedCreate = `.create-merge table ['mytable'] (['fields']:dynamic, ['name']:string, ['tags']:dynamic, ['timestamp']:datetime);`
-	const expectedMapping = `` +
-		`.create-or-alter table ['mytable'] ingestion json mapping 'mytable_mapping' '[{"column":"fields", ` +
-		`"Properties":{"Path":"$[\'fields\']"}},{"column":"name", "Properties":{"Path":"$[\'name\']"}},{"column":"tags", ` +
-		`"Properties":{"Path":"$[\'tags\']"}},{"column":"timestamp", "Properties":{"Path":"$[\'timestamp\']"}}]'`
-	require.Equal(t, expectedCreate, createTableCommand(tableName).String())
-	require.Equal(t, expectedMapping, createTableMappingCommand(tableName).String())
+func TestConnectBlankEndpointData(t *testing.T) {
+	plugin := AzureDataExplorer{
+		Log:    testutil.Logger{},
+		Client: &adx_common.Client{},
+		Config: adx_common.Config{
+			Endpoint: "",
+		},
```

Comment:
```suggestion
```

Reply:
It _is_ empty if you don't set it as Golang initializes structure fields with the `nil` value of the type and that's and empty string for strings.